tools/jdbc/src/main/java/org/duckdb/DuckDBArrayResultSet.java

```diff
@@ -0,0 +1,1044 @@
+package org.duckdb;
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.sql.*;
+import java.util.Calendar;
+import java.util.Map;
+
+public class DuckDBArrayResultSet implements ResultSet {
+
+    private DuckDBVector vector;
+    int offset, length;
+
+    int currentValueIndex = -1;
+    boolean closed = false;
+    boolean wasNull = false;
+
+    public DuckDBArrayResultSet(DuckDBVector vector, int offset, int length) {
+        this.vector = vector;
+        this.offset = offset;
+        this.length = length;
```

Comment:
Looks like you haven't used this offset anywhere, this means this will break on multiple arrays being returned by a single query. Easiest fix is to just use the methods on the original Array implementation

Reply:
@Mause Thank you! I was originally hoping to wrap `DuckDBArray` instead of `DuckDBVector`, but I only found an untyped `Object getArray()` on it. Reusing vector seemed less error-prone than type-checking in every getter. Are there better ways of using `DuckDBArray` that I missed?
 
In the meantime, I updated `DuckDBArrayResultSet` to use the offset. I could not find a test case to force a non-zero offset -- do you know offhand what query would work to test this case?

Reply:
Something like
```sql
SELECT UNNEST([[1], [2]])
```
Should give a non zero offset, by having two list rows for a single column

Reply:
Thank you! That worked to expose the offset issue on the previous implementation of `getValue()`. I added the test.